core/scripts/cli/sponsor/currencyUtils.js

```diff
@@ -14,7 +14,23 @@ const unwrapToEth = async (web3, artifacts, emp, amount) => {
   await submitTransaction(web3, async () => await weth.withdraw(amount.toString()), "Unwrapping WETH to ETH");
 };
 
+const getIsWeth = async (web3, artifacts, collateralCurrency) => {
+  const WETH9 = artifacts.require("WETH9");
+  return collateralCurrency.address === WETH9.address;
```

Comment:
Doesn't this address get set by the deployment? But the contract is already deployed in this case?

Reply:
IIUC, the deployment sets WETH9 to point to the actual WETH contract in prod, and deploys a custom one only for local. I think this check should work correctly, but I haven't tested it against prod or anything yet.

Reply:
@ptare I think you are right here as the migration is skipped if there is a `wethAddress` for the given network. After running `apply-registry` the build files are updated with the addresses specified in the `core/networks` directory. However this only looks like network 42 has this specified, which is kovan. the other networks dont have the WETH9 added. The networks are specified correctly in `common/PublicNetworks.js` but this is not what `apply-registry` pulls from.

I was able to verify this by running a truffle console on kovan and `WETH9.address` returned the correctly deployed address however all other addresses are missing this address as they are not specified in the networks tab. I created a [PR](https://github.com/UMAprotocol/protocol/pull/1154) to solve this by adding the appropriate WETH addresses to the other networks.

Reply:
Isn't that because we haven't migrated to those networks yet? I don't fully understand how this is all hooked up, so I could be totally wrong.